features/device_key/source/DeviceKey.h

```diff
@@ -139,16 +139,16 @@ class DeviceKey : private mbed::NonCopyable<DeviceKey> {
     int get_derived_key(uint32_t *ikey_buff, size_t ikey_size, const unsigned char *isalt, size_t isalt_size,
                         unsigned char *output, uint32_t ikey_type);
 
-    /** Generate a random ROT key by using entropy
-     * @param output Output buffer for the generated key.
-     * @param size Input: The size of the buffer. If size is less
-     *                    than 16 bytes, the method generates an
-     *                    error. 16-31 bytes creates a 16-byte key.
-     *                    32 or higher generates a 32-byte key
-     *             Output: The actual written size to the buffer
-     * @return 0 on success, negative error code on failure
+    /** Generate Root of Trust from random sources.
+     * Uses TRNG or various other entropy sources to generate random device key and
+     * inject that into device's KVStore. Device Key can only be generated once.
+     *
+     * \return DEVICEKEY_SUCCESS, when device key succesfully generated and injected.
+     * \return DEVICEKEY_ALREADY_EXIST, if the key has already been written.
+     * \return DEVICEKEY_GENERATE_RANDOM_ERROR if this device does not contain entropy sources and cannot generate a key.
+     * \return error codes on other failures.
      */
-    int generate_key_by_random(uint32_t *output, size_t size);
+    int generate_root_of_trust();
```

Comment:
Should be changed to public api. Seems like it could/should also be static.

Reply:
Writing  a unit test for this wouldn't hurt.

Reply:
It is now changed to public.

  static DeviceKey &get_instance()
    {
        // Use this implementation of singleton (Meyer's) rather than the one that allocates
        // the instance on the heap, as it ensures destruction at program end (preventing warnings
        // from memory checking tools, such as valgrind).
        static DeviceKey instance;
        return instance;
    }